### PR TITLE
[Fix] refresh when truncate table

### PR DESCRIPTION
**Pull request by IHEII**
## Summary

- Fix refresh table entry when the table has been truncated

## Solution Description

- Fix refresh table entry when the table has been truncated

### CODE REVIEW

File: src/main/java/com/alipay/oceanbase/rpc/ObTableClient.java [hunk 1]
```diff
@@ -1389,7 +1427,7 @@ public ObPair<Long, ObTableParam> getTable(String tableName, List<ObNewRange> ke
     /**
      * get addr by pardId
      * @param tableName table want to get
-     * @param partId partId where table located
+     * @param partId partId where table located (should be partIdx)
```
Comment (shenyunlong) [inline src/main/java/com/alipay/oceanbase/rpc/ObTableClient.java:1430]: is it more understandable to rename the `partId` parameter to `partIdx`

File: src/main/java/com/alipay/oceanbase/rpc/ObTableClient.java [hunk 2]
```diff
@@ -1021,6 +1021,21 @@ public ObIndexInfo getOrRefreshIndexInfo(final String indexName, final String in
...
+     * Get or refresh table entry.
+     * @param tableName table name
+     * @param refresh is re-fresh
+     * @param waitForRefresh wait re-fresh
+     * @param fetchAll fetch all data from server if needed
+     * @return this
+     * @throws Exception if fail
+     */
+    public TableEntry getOrRefreshTableEntry(final String tableName, final boolean refresh,
+                                             final boolean waitForRefresh, boolean fetchAll)
```
[the remark this answers is not in the document]
Comment (IHEII) [reply src/main/java/com/alipay/oceanbase/rpc/ObTableClient.java:1037]: **fetchAll** is true only if the client has continuously failed to execute on a table for _runtime.continuous.failure.ceiling_ times.

File: src/main/java/com/alipay/oceanbase/rpc/ObTableClient.java (hunk 1 above)
Comment (IHEII) [reply src/main/java/com/alipay/oceanbase/rpc/ObTableClient.java]: Actually, it is partition id in 3.x, which is different from partIdx and tablet id in 4.x. So I have commented on it again.

**Comment by shenyunlong** [comment]
LGTM

**Pull request merged**